lib/puppet/provider/cisco_overlay_global/nxapi.rb

```diff
@@ -41,60 +41,44 @@
     :dup_host_mac_detection_timeout,
   ]
 
-  PuppetX::Cisco::AutoGen.mk_puppet_methods(:non_bool, self, '@vxlan_global',
-                                            VXLAN_GLOBAL_PROPS)
+  PuppetX::Cisco::AutoGen.mk_puppet_methods(:non_bool, self, '@overlay_global',
+                                            OVERLAY_GLOBAL_PROPS)
 
   def initialize(value={})
     super(value)
-    @vxlan_global = Cisco::VxlanGlobal.new if Cisco::Feature.fabric_forwarding_enabled?
+    @overlay_global = Cisco::OverlayGlobal.new
     @property_flush = {}
   end
 
-  def self.properties_get(vxlan_global)
-    current_state = {
-      ensure: :present,
-      name:   'default',
-    }
+  def self.properties_get(overlay_global)
```

Comment:
In a similar vein, we're using `nu_obj` here. 

No need to change these now, just letting you know.